src/api/client_server/keys.rs

```diff
@@ -338,7 +339,15 @@ pub(crate) async fn get_keys_helper<F: Fn(&UserId) -> bool>(
 			hash_map::Entry::Vacant(e) => {
 				e.insert((Instant::now(), 1));
 			},
-			hash_map::Entry::Occupied(mut e) => *e.get_mut() = (Instant::now(), e.get().1 + 1),
+			hash_map::Entry::Occupied(mut e) => {
+				*e.get_mut() = (
+					Instant::now(),
+					e.get()
+						.1
+						.checked_add(1)
+						.expect("bad_query_ratelimiter attempt/try count should not ever get this high"),
```

Comment:
Does it make sense to potentially crash here instead of just using .saturating_add()? As far as I can tell the restart counter is only used to calculate the exponential backoff time, which is limited by MAX_DURATION anyway.

Reply:
Switched it to saturating add, yes you are right this is limited by exponential backoff